src/Simulation/QCTraceSimulator/QubitMetrics.cs

```diff
@@ -8,14 +8,27 @@ namespace Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime
     /// </summary>
     public class QubitTimeMetrics
     {
-        // TODO: Qubit Ids are already available in qubits, but DepthCounter doesn't have access to it
-        // in OnPrimitiveOperation because it's not part of IQCTraceSimulatorListener interface.
+        // Qubit Ids are already available in qubits, but DepthCounter doesn't have access to them
+        // in OnPrimitiveOperation because they are not part of IQCTraceSimulatorListener interface.
         // Consider changing architecture to pass qubits rather than metrics in IQCTraceSimulatorListener.
         public long QubitId { get; }
 
+        // If qubit is not fixed in time, this is depth of single-qubit-gate chain on this qubit
+        // If qubit is fixed in time, this is the busy interval end time for this qubit
+        internal ComplexTime EndTime { get; set; }
+
+
+        // If qubit is not fixed in time, this is ComplexTime.MinValue
+        // If qubit is fixed in time, this is the busy interval start time
+        internal ComplexTime StartTime { get; set; }
```

Comment:
Would it help to add a method like `bool IsFixedInTime() { /*validate that the settings on start/end are consistent and return whether the qubit is fixed in time based on the current values of the fields*/ }` to both document here and use in DepthCounter class?

Reply:
It is only used in the context when StartTime is adjusted as a result of comparison so it will make this more obscure. With the updated comments I think it's clear what's going on.